code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityExtensionTests.java

```diff
@@ -629,39 +629,41 @@ public void test_processCachedEvents_processesWhenBooted() {
 	// ========================================================================================
 	// handleRequestContent
 	// ========================================================================================
-	// use callback to see if sharedstate is set properly
-	// similar to other ones where update identities
-	// if calling updateAdId directly, already passing in a callback
+
 	@Test
 	public void test_handleRequestContent_processesWhenBooted() {
-		Map<String, Object> identityXDM = createXDMIdentityMap(new TestItem("space", "moon"));
-		MockIdentityState mockIdentityState = new MockIdentityState(new IdentityProperties(identityXDM));
-		mockIdentityState.hasBooted = true;
-		extension.state = mockIdentityState;
-
-		// test
-		extension.processAddEvent(buildUpdateIdentityRequest(identityXDM));
-		extension.processAddEvent(buildRemoveIdentityRequest(identityXDM));
-		extension.processAddEvent(
-			new Event.Builder(
-				"Test event",
-				IdentityConstants.EventType.EDGE_IDENTITY,
-				IdentityConstants.EventSource.REQUEST_IDENTITY
-			)
-				.build()
-		);
-		extension.processAddEvent(
-			new Event.Builder(
-				"Test event",
-				IdentityConstants.EventType.GENERIC_IDENTITY,
-				IdentityConstants.EventSource.REQUEST_RESET
+		// Setup
+		final Event event = new Event.Builder(
+			"Test Ad ID event",
+			IdentityConstants.EventType.GENERIC_IDENTITY,
+			IdentityConstants.EventSource.REQUEST_CONTENT
+		)
+			.setEventData(
+				new HashMap<String, Object>() {
+					{
+						put(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER, "adId");
+					}
+				}
 			)
-				.build()
-		);
+			.build();
 
-		// verify
-		verify(mockExtensionApi, times(3))
-			.setXDMSharedEventState(any(Map.class), any(Event.class), any(ExtensionErrorCallback.class)); // request identity does not update shared state
+		final ArgumentCaptor<Map> sharedStateCaptor = ArgumentCaptor.forClass(Map.class);
+		final ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+
+		// Test
+		extension.handleRequestContent(event);
+
+		// Verify
+		verify(mockExtensionApi, times(1))
+			.setXDMSharedEventState(sharedStateCaptor.capture(), eq(event), any(ExtensionErrorCallback.class));
+		PowerMockito.verifyStatic(MobileCore.class, Mockito.times(1));
+		MobileCore.dispatchEvent(eventCaptor.capture(), any(ExtensionErrorCallback.class));
```

Comment:
Can you use the `eventCaptor` to verify the structure and value of the Consent event?

Reply:
Added consent event verification using the event captor